external-crates/move/move-compiler/src/expansion/translate.rs

```diff
@@ -2674,7 +2692,16 @@ fn assign(context: &mut Context, sp!(loc, e_): P::Exp) -> Option<E::LValue> {
             let en = name_access_chain(context, Access::ApplyNamed, pn)?;
             let tys_opt = optional_types(context, ptys_opt);
             let efields = assign_unpack_fields(context, loc, pfields)?;
-            EL::Unpack(en, tys_opt, efields)
+            EL::Unpack(en, tys_opt, E::FieldBindings::Named(efields))
+        }
+        PE::Call(pn, false, ptys_opt, sp!(_, exprs)) => {
+            context
+                .env
+                .check_feature(FeatureGate::PositionalFields, context.current_package, loc);
```

Comment:
Thoughts on doing all feature gating like this during expansion? 

Just want to avoid having to feature gating across different passes for the same feature

Reply:
Based on the fact that we have to do a check in naming. Probably best to gate  as soon as we can for each feature, as you have done here